sensor/common/filesystem/pipeline/pipeline.go

```diff
@@ -0,0 +1,175 @@
+package pipeline
+
+import (
+	"context"
+
+	sensorAPI "github.com/stackrox/rox/generated/internalapi/sensor"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/protocompat"
+	"github.com/stackrox/rox/pkg/uuid"
+	"github.com/stackrox/rox/sensor/common/clusterentities"
+	"github.com/stackrox/rox/sensor/common/detector"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+type Pipeline struct {
+	detector detector.Detector
+	stopper  concurrency.Stopper
+
+	accessChan      chan *storage.FileAccess
+	clusterEntities *clusterentities.Store
+
+	msgCtx context.Context
+}
+
+func NewFileSystemPipeline(detector detector.Detector, clusterEntities *clusterentities.Store) *Pipeline {
+	msgCtx := context.Background()
+
+	p := &Pipeline{
+		detector:        detector,
+		accessChan:      make(chan *storage.FileAccess),
```

Comment:
This channel is never closed. I'm aware we follow this pattern in many places in sensor but we are trying to move away from it. I'd suggest to pass this channel as a parameter and handle it's life-cycle in the producer of the event (in this case I believe it should be the filesystem service) and then in the service instead of calling `Process` just write in that channel. The `translate` call will have to be moved to after we read from the channel in the `run` goroutine.

Reply:
I now pass the channel to the pipeline. I also added
```
func (s *serviceImpl) Stop() {
        s.pipeline.Shutdown()
}
```

to the service. `pipeline.Shutdown()` closes the channel. However, I am not sure that `Stop` is ever called.

Reply:
I am not 100% clear on the rest of what you are saying. The channel reads in `*storage.FileAcess`, but the service has messages of type `sensorAPI.FileActivity`. Should the translation be done in the service? Should the type of the channel be changed?